src/material-experimental/mdc-select/harness/select-harness.ts

```diff
@@ -0,0 +1,122 @@
+  private _documentRootLocator = this.documentRootLocatorFactory();
+  private _panel = this._documentRootLocator.locatorFor(PANEL_SELECTOR);
+  private _backdrop = this._documentRootLocator.locatorFor('.cdk-overlay-backdrop');
+  private _optionalPanel = this._documentRootLocator.locatorForOptional(PANEL_SELECTOR);
+  private _options = this._documentRootLocator.locatorForAll(`${PANEL_SELECTOR} .mat-option`);
+  private _groups = this._documentRootLocator.locatorForAll(`${PANEL_SELECTOR} .mat-optgroup`);
+  private _trigger = this.locatorFor('.mat-select-trigger');
+  private _value = this.locatorFor('.mat-select-value');
+
+  static hostSelector = '.mat-select';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a select with
+   * specific attributes.
+   * @param options Options for narrowing the search.
+   * @return `HarnessPredicate` configured with the given options.
+   */
+  static with(options: SelectHarnessFilters = {}): HarnessPredicate<MatSelectHarness> {
+    return new HarnessPredicate(MatSelectHarness)
+        .addOption('id', options.id, async (harness, id) => {
+          const harnessId = (await harness.host()).getAttribute('id');
+          return (await harnessId) === id;
+        });
+  }
+
+  /** Gets a boolean promise indicating if the select is disabled. */
+  async isDisabled(): Promise<boolean> {
+    return (await this.host()).hasClass('mat-select-disabled');
+  }
+
+  /** Gets a boolean promise indicating if the select is valid. */
+  async isValid(): Promise<boolean> {
+    return !(await (await this.host()).hasClass('ng-invalid'));
+  }
+
+  /** Gets a boolean promise indicating if the select is required. */
+  async isRequired(): Promise<boolean> {
+    return (await this.host()).hasClass('mat-select-required');
+  }
+
+  /** Gets a boolean promise indicating if the select is empty (no value is selected). */
+  async isEmpty(): Promise<boolean> {
+    return (await this.host()).hasClass('mat-select-empty');
+  }
+
+  /** Gets a boolean promise indicating if the select is in multi-selection mode. */
+  async isMultiple(): Promise<boolean> {
+    const ariaMultiselectable = (await this.host()).getAttribute('aria-multiselectable');
+    return (await ariaMultiselectable) === 'true';
+  }
+
+  /** Gets a promise for the select's value text. */
+  async getValueText(): Promise<string> {
+    return (await this._value()).text();
+  }
+
+  /** Focuses the select and returns a void promise that indicates when the action is complete. */
+  async focus(): Promise<void> {
+    return (await this.host()).focus();
+  }
+
+  /** Blurs the select and returns a void promise that indicates when the action is complete. */
+  async blur(): Promise<void> {
+    return (await this.host()).blur();
+  }
+
+  /** Gets the select panel. */
+  async getPanel(): Promise<TestElement> {
+    return this._panel();
+  }
+
+  /** Gets the options inside the select panel. */
+  async getOptions(): Promise<TestElement[]> {
```

Comment:
We could potentially expose an `OptionHarness` and `OptionGroupHarness` to make it easier to work with these. I'm not sure if its overkill for this case, but I kind of like the idea of avoiding giving the user `TestElement` references when possible.

I would at least add an optional parameter to these methods so I can do something like `getOptionGroups('Food')` or `getOptions('Pizza')`. If you make a `OptionHarness` this could be an `OptionHarnessFilters` instead of a string.

Curious if @jelbourn has an opinion here

Reply:
We can do it on the second pass once both the autocomplete and select harnesses get in since both of them use `mat-option`.

Reply:
Agreed, I think we should have harnesses for the options. It can go in a follow-up PR.

Reply:
Add `TODO` for now